### ISSUE TITLE
while_executing working wrong

### ISSUE THREAD

**Issue by HienNguyenAsnet** [mhenrixon/sidekiq-unique-jobs#193]
As description, it is to make sure that a job can be scheduled any number of times but only executed a single time per argument provided to the job we call this runtime uniqueness. But it not work for me

```
class Job
  include Sidekiq::Worker
  sidekiq_options queue: :test, retry: true, unique: :while_executing

  def perform(arg)
    sleep 5.minutes
  end
end

Job.perform_async(1)
Job.perform_async(1)
Job.perform_async(2)
```

Have two `Job` with arg is `1` execute at the time

<img width="1280" alt="screen shot 2016-08-29 at 17 31 24" src="https://cloud.githubusercontent.com/assets/5277254/18048787/8174c51c-6e0e-11e6-974b-5b4909a5a2aa.png">

System information
- redis: `3.3.1`
- sidekiq: `4.1.3`
- sidekiq-unique-jobs: `4.0.18`
- sidekiq-limit_fetch (https://github.com/brainopia/sidekiq-limit_fetch#limits): `3.1.0`

Sidekiq Config

```
:queues:
  - test
  - default
:limits:
  test: 3
```

**Comment by natematykiewicz** [comment]
Have you tried it without `sidekiq-limit_fetch`? Its says that it's incompatible with "any other plugin that rewrites fetch strategy of sidekiq.".

This limit fetch gem is going to try to control when jobs are performed from the queue, but that's exactly what this gem is needs to do for the `while_executing` option.

**Comment by HienNguyenAsnet** [comment]
@natematykiewicz Thank for feedback.

I just check it without sidekiq-limit_fetch. Still have two Job with arg is 1 execute at the time.

**Comment by HienNguyenAsnet** [comment]
Below is list gem relate with sidekiq I used
```
gem 'sidekiq', '~> 4.1', '>= 4.1.2'
gem 'sidekiq-cron', '~> 0.4.0'
gem 'sidekiq-failures'
# gem 'sidekiq-limit_fetch'
gem 'sidekiq-statistic'
gem 'sidekiq-unique-jobs'
```

**Comment by natematykiewicz** [comment]
If I were you, I'd remove all gems except sidekiq and sidekiq-unique-jobs and then see what you get. If it works, start adding gems in until you find one that doesn't seem to be compatible with sidekiq-unique-jobs.

**Comment by HienNguyenAsnet** [comment]
```
gem 'sidekiq', '~> 4.1', '>= 4.1.2'
# gem 'sidekiq-cron', '~> 0.4.0'
# gem 'sidekiq-failures'
# gem 'sidekiq-limit_fetch'
# gem 'sidekiq-statistic'
gem 'sidekiq-unique-jobs'
```
It's same result, still have two Job with arg is 1 execute at the time.

**Comment by aserafin** [comment]
I had the same experience - the only option that seems to work for me is `until_executed`

**Comment by cheneveld** [comment]
I was having some issues with this. I initially had one worker process with 2 queues:

```worker: env DB_POOL=$WORKER_CONCURRENCY bundle exec sidekiq -q default -q keep_em_cached -c $WORKER_CONCURRENCY```

And moved my unique jobs on its own process

```
worker: env DB_POOL=$WORKER_CONCURRENCY bundle exec sidekiq -q default -q keep_em_cached -c $WORKER_CONCURRENCY
another_worker: env DB_POOL=4 bundle exec sidekiq -q another_queue -c 4
```

This seemed to solve my unique issues.  I didn't investigate any more than this so, I can't say what was actually happening.

**Comment by mhenrixon** [comment]
Thanks for the report @cheneveld

**Comment by HienNguyenAsnet** [comment]
Sorry, I'm pretty busy recent time.

@cheneveld, @mhenrixon Tried split 2 process but no luck. It's same result.

**Comment by ryanzbone** [comment]
I'm seeing the same thing with sidekiq enterprise. I have following in my worker:

```ruby
sidekiq_options unique: :while_executing, uniqe_args: ->(args) { [args.first, args.second] }
```
But I end up with this:
![screen shot 2016-10-27 at 2 46 33 pm](https://cloud.githubusercontent.com/assets/2286261/19780734/3edd6528-9c54-11e6-8604-3b0dcefdc40a.png)

**Comment by untoldone** [comment]
@mhenrixon This seems like a pretty core use case to be broken? Super simple repro:

```
class TestWorker
  include Sidekiq::Worker

  sidekiq_options unique: :while_executing

  def perform(one, two)
    sleep 10
  end
end
```

And then start with
```
TestWorker.perform_async 1, 2
TestWorker.perform_async 1, 2
TestWorker.perform_async 1, 2
TestWorker.perform_async 1, 2
TestWorker.perform_async 1, 2
```

Am I missing something here?

**Comment by jasonbuehler** [comment]
I am also seeing this behavior. Is while_executing option broken? It doesn't appear to lock other jobs out, everything with the same args enqueues all at once.

**Comment by mhenrixon** [comment]
It has been somewhat fixed in v5.0.5. Check the readme for how it currently works: https://github.com/mhenrixon/sidekiq-unique-jobs#while-executing
